homeassistant/components/tplink/config_flow.py

```diff
@@ -137,16 +320,111 @@ def _async_create_entry_from_device(self, device: SmartDevice) -> FlowResult:
             title=f"{device.alias} {device.model}",
             data={
                 CONF_HOST: device.host,
+                CONF_ALIAS: device.alias,
+                CONF_MODEL: device.model,
+                CONF_DEVICE_CONFIG: device.config.to_dict(
+                    credentials_hash=device.credentials_hash,
+                    exclude_credentials=True,
+                ),
             },
         )
 
+    async def _async_try_discover_and_update(
+        self,
+        host: str,
+        credentials: Optional[Credentials],
+        raise_on_progress: bool,
+    ) -> SmartDevice:
+        """Try to connect."""
```

Comment:
The docstring is a bit misleading, as it does something more than just a connection try.

Reply:
Updated